### PR TITLE
Remove Joda dependency

### PR DESCRIPTION
**Pull request by daschult**

### CODE REVIEW

File: azure-client-authentication/src/main/java/com/microsoft/azure/v2/credentials/AzureCliToken.java [hunk 1]
```diff
@@ -72,11 +71,7 @@ String accessToken() {
 
     Date expiresOn() {
```
Comment (RikkiGibson) [inline azure-client-authentication/src/main/java/com/microsoft/azure/v2/credentials/AzureCliToken.java]: Weird. I guess this class won't be in our library when the new standalone auth library is released, at least.

File: client-runtime/src/main/java/com/microsoft/rest/v2/SwaggerMethodParser.java [hunk 2]
```diff
@@ -43,6 +46,7 @@
  * method.
  */
 public class SwaggerMethodParser {
+    private final SerializerAdapter<?> serializer;
```
Comment (RikkiGibson) [inline client-runtime/src/main/java/com/microsoft/rest/v2/SwaggerMethodParser.java]: Is this for handling template substitutions on non-String parameters?

File: azure-client-authentication/src/main/java/com/microsoft/azure/v2/credentials/AzureCliToken.java (hunk 1 above)
Comment (daschult) [reply azure-client-authentication/src/main/java/com/microsoft/azure/v2/credentials/AzureCliToken.java]: Yeah. I left this one as Date on purpose.

File: client-runtime/src/main/java/com/microsoft/rest/v2/SwaggerMethodParser.java (hunk 2 above)
Comment (daschult) [reply client-runtime/src/main/java/com/microsoft/rest/v2/SwaggerMethodParser.java]: Correct. Specifically non-String header values and non-String query values.

File: client-runtime/src/main/java/com/microsoft/rest/v2/serializer/DateTimeSerializer.java [hunk 3]
```diff
@@ -29,17 +29,33 @@
...
+     * Convert the provided OffsetDateTime to its String representation.
+     * @param offsetDateTime The OffsetDateTime to convert.
+     * @return The String representation of the provided offsetDateTime.
+     */
+    public static String toString(OffsetDateTime offsetDateTime) {
+        String result = null;
+        if (offsetDateTime != null) {
+            offsetDateTime = offsetDateTime.withOffsetSameInstant(ZoneOffset.UTC);
+            result = DateTimeFormatter.ISO_INSTANT.format(offsetDateTime);
+            if (result.startsWith("+")) {
```
Comment (RikkiGibson) [inline client-runtime/src/main/java/com/microsoft/rest/v2/serializer/DateTimeSerializer.java]: Welp

Comment (RikkiGibson) [reply client-runtime/src/main/java/com/microsoft/rest/v2/serializer/DateTimeSerializer.java]: Has anyone out in the wild reported/discussed/rationalized this behavior?

Comment (daschult) [reply client-runtime/src/main/java/com/microsoft/rest/v2/serializer/DateTimeSerializer.java]: The ISO 8601 standard says that outside of 4 digit years any two parties using these serialized strings can determine whether they want to use + in front of their years. Joda and Autorest didn't use + signs, but it looks like the Java OffsetDateTime does use it. I'd rather not, so I just strip the leading + if it exists.

**Comment by RikkiGibson** [comment]
Seems all good to me--too bad we had to roll own implementations of a couple more conversions.

**Pull request merged**